Tasks/WindowsMachineFileCopy/WindowsMachineFileCopy.ps1

```diff
@@ -1,122 +1,108 @@
-$targetPath = $targetPath.Trim('"')
-
-# Default + constants #
-$resourceFQDNKeyName = Get-ResourceFQDNTagKey
-
-$envOperationStatus = 'Passed'
-
-Validate-SourcePath $sourcePath
-Validate-DestinationPath $targetPath $environmentName
-
-if([string]::IsNullOrWhiteSpace($environmentName))
-{
-    Write-Verbose "No environment found. Copying to destination."
-
-    Write-Output (Get-LocalizedString -Key "Copy started for - '{0}'" -ArgumentList $targetPath)
-    Copy-OnLocalMachine -sourcePath $sourcePath -targetPath $targetPath -adminUserName $adminUserName -adminPassword $adminPassword `
-                        -cleanTargetBeforeCopy $cleanTargetBeforeCopy -additionalArguments $additionalArguments
-    Write-Verbose "Files copied to destination successfully."
-}
-else
+try 
 {
 
-    $connection = Get-VssConnection -TaskContext $distributedTaskContext
-
-    Write-Verbose "Starting Register-Environment cmdlet call for environment : $environmentName with filter $machineFilter"
-    $environment = Register-Environment -EnvironmentName $environmentName -EnvironmentSpecification $environmentName -UserName $adminUserName -Password $adminPassword -Connection $connection -TaskContext $distributedTaskContext -ResourceFilter $machineFilter
-    Write-Verbose "Completed Register-Environment cmdlet call for environment : $environmentName"
+    # keep machineNames parameter name unchanged due to back compatibility
+    $machineFilter = $machineNames
+    $sourcePath = $sourcePath.Trim('"')
+    $targetPath = $targetPath.Trim('"')
 
-    $fetchedEnvironmentName = $environment.Name
+    $envOperationStatus = 'Passed'
 
-    Write-Verbose "Starting Get-EnvironmentResources cmdlet call on environment name: $fetchedEnvironmentName"
-    $resources = Get-EnvironmentResources -Environment $environment
-    Write-Verbose "Completed Get-EnvironmentResources cmdlet call for environment name: $fetchedEnvironmentName"
+    Validate-SourcePath $sourcePath
+    Validate-DestinationPath $targetPath $environmentName
 
-    if ($resources.Count -eq 0)
+    if([string]::IsNullOrWhiteSpace($environmentName))
     {
-         throw (Get-LocalizedString -Key "No machine exists under environment: '{0}' for deployment" -ArgumentList $environmentName)
+        Write-Verbose "No environment found. Copying to destination."
+
+        Write-Output (Get-VstsLocString -Key "Copy started for - '{0}'" -ArgumentList $targetPath)
+        Copy-OnLocalMachine -sourcePath $sourcePath -targetPath $targetPath -adminUserName $adminUserName -adminPassword $adminPassword `
+                            -cleanTargetBeforeCopy $cleanTargetBeforeCopy -additionalArguments $additionalArguments
+        Write-Verbose "Files copied to destination successfully."
     }
+    else
+    {
 
-    $resourcesPropertyBag = Get-ResourcesProperties -envName $fetchedEnvironmentName -resources $resources
+        $machines = $environmentName.split(',') | ForEach-Object { if ($_ -and $_.trim()) { $_.trim() } }
```

Comment:
will be better if we log for machine names like - copy will be done for machines - m1,m2,m3